ui/app/components/tool-wrap.js

```diff
@@ -0,0 +1,27 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class HashTool extends Component {
+  @tracked data = '{\n}';
+  @tracked buttonDisabled = false;
+
+  @action
+  onClear() {
+    this.args.onClear();
+  }
+  @action
+  updateTtl(evt) {
+    if (!evt) return;
+    const ttl = evt.enabled ? `${evt.seconds}s` : '30m';
+    this.args.updateTtl(ttl);
+  }
+  @action
+  codemirrorUpdated(val, codemirror) {
+    codemirror.performLint();
+    const hasErrors = codemirror.state.lint.marked.length > 0;
```

Comment:
I wonder if we should use optional chaining here in case `marked` or anything upstream comes back `undefined`?

Reply:
Good point.